### PR TITLE
Add documentation and submodules status

### PR DESCRIPTION
**Pull request by micaelmbagira**
- Add submodule status
- Update CONTRIBUTING
- Add issue and PR templates
- Add USAGE

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+
+>* **Does this PR introduce a breaking change ?** (What changes might users need to make in their application due to this PR?)
+
+
+
+>* **Other information**:
+
+
+## <a name="build"></a> Build tests
```
Comment (zckrs) [inline CONTRIBUTING.md:152]: Why `<a name="build"></a>` ?

Comment (micaelmbagira) [inline CONTRIBUTING.md:152]: This is an anchor for linked to https://github.com/FountainJS/fountain/pull/28/commits/3945e969c6ee6d55f31d3cd2f81a21eb13dff285#diff-6a3371457528722a734f3c51d9238c13R9

Comment (zckrs) [inline CONTRIBUTING.md:152]: I think github will auto add anchor #build-tests

But maybe this doc will appear on fountainjs.io

Comment (Swiip) [inline CONTRIBUTING.md:152]: It will ^^

File: USAGE.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
+Usage
+=====
+
+## Requirements
+
+You need at least Node `v4.0.0` and npm `2`.
+
+## Create your project
+Install the required tools: yo, gulp, bower
```
Comment (Swiip) [inline USAGE.md]: Inconsistency regarding the line bellow

Comment (Swiip) [inline USAGE.md]: I would add a small sentence like "You only need the tool corresponding with the technologies you'll choose with the generator" (don't want to scare a Babel guy with an install of typings)

File: USAGE.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+* `--css` Description for css preprocessor
+* `--js` Description for js preprocessor
+
+## Use Gulp tasks
+
+- gulp or gulp build to build an optimized version of your application in /dist
+- gulp serve to launch a browser sync server on your source files
+- gulp serve:dist to launch a server on your optimized application
+- gulp test to launch your unit tests with Karma
+- gulp test:auto to launch your unit tests with Karma in watch mode
```
Comment (Swiip) [inline USAGE.md]: backquotes for gulp commands would be nicer

File: USAGE.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+## Directory structure
+
+Every component of the app get its own directory which contains the component code (depending on the framework, it can be `*.(js|ts)` or/and `*.component.(js|ts)` or/and `*.controller.(js|ts)` or/and `*.service.(js|ts)` or/and `*.directive.(js|ts)` and `*.spec.(js|ts)`.
```
Comment (Swiip) [inline USAGE.md]: That's obsolete talking about directives ^^ Even for ng1 we removed `.component.js` files suffixes

File: USAGE.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+│   │   │   └──  techs.spec.(js|ts)
+│   │   │
+│   │   ├── footer.(js|ts)
+│   │   ├── footer.spec.(js|ts)
+│   │   ├── header.(js|ts)
+│   │   ├── header.spec.(js|ts)
+│   │   ├── main.(js|ts)
+│   │   ├── main.spec.(js|ts)
+│   │   ├── title.(js|ts)
+│   │   └── title.spec.(js|ts)
```
Comment (Swiip) [inline USAGE.md:106]: It implies the use of the Tech template. We can't put the directory structure of both Tech and Hello but perhaps put a warning on top.

File: USAGE.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+├──  bower.json (optional)
+├──  gulpfile.js
+├──  jspm.config.js (optional)
+├──  jspm.browser.js (optional)
+└──  package.json
+</pre>
+
+## Configuration
+
+You can file the configuration files in `conf/`.
```
Comment (Swiip) [inline USAGE.md]: find ?

File: USAGE.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+
+
+## Questions the generator will ask
+
+- *Framework*: React, Angular 1, Angular 2
+- *Modules manager*: Webpack with NPM, SystemJS with JSPM, None with Bower injection
+- *CSS preprocessor*: SASS, Less, CSS
+- *JS preprocessor*: ES2015 with Babel, Pure old JavaScript, TypeScript
+- *Sample app*: A working landing page, A simple Hello Word, a Redux TodoApp
```
Comment (Swiip) [inline USAGE.md]: TodoMVC !

Or perhaps "for combination where it's available, an corresponding implementation of TodoMVC"

File: USAGE.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+You can file the configuration files in `conf/`.
+
+* *browsersync.conf(-dist).js*: define which folder to serve
+* *gulp.conf.js*: contains the variables used in other gulp files which defines tasks
+* *karma(-auto).conf.js*: contains Karma configuration for unit tests
+* *ts.conf.js*: TypeScript compiler configuration regarding the framework used
+* *tslint.conf.json*: TypeScript linter rules
+* *webpack(-dist | test).js*: Webpack configuration with definition of loaders and plugins regarding the environment (dev, dist or test)
+
+## Gulpfile
```
Comment (Swiip) [inline USAGE.md:134]: Proposition:

As the `gulp_tasks` folder contains all gulp tasks used by the project, the `gulpfile.js` centralize all processes by chaining the actual tasks.

It's made possible by the new Gulp 4 which allow to define precise task series and parallelization.

File: USAGE.md (hunk 7 above)
Comment (zckrs) [inline USAGE.md]: I would prefer Boilerplate instead of Sample app like html5-boilerplate, https://github.com/sindresorhus/electron-boilerplate etc

Comment (Swiip) [inline USAGE.md]: I'm not sure. Boilerplate means "something boring but that you need" to me. But here we talk about an app we give as an example that you have to remove to put your own.

**Comment by zckrs** [comment]
Some awesome issue/PR template: https://github.com/devspace/awesome-github-templates

File: USAGE.md (hunk 2 above)
Comment (micaelmbagira) [inline USAGE.md]: 👍

File: USAGE.md (hunk 3 above)
Comment (micaelmbagira) [inline USAGE.md]: 👍

File: USAGE.md (hunk 4 above)
Comment (micaelmbagira) [inline USAGE.md]: 👍

File: USAGE.md (hunk 5 above)
Comment (micaelmbagira) [inline USAGE.md:106]: 👍

**Pull request closed without merging**